src/field.ts

```diff
@@ -112,7 +112,9 @@ export class FieldImplementation<T, Model>
             value: this.value[key],
             originalValue: this.value[key],
             onUpdate: () => {
-              this.value[key] = field.value;
+              if (this.value) {
```

Comment:
this looks weird to me. 
I expect this to work:
```ts
fields.phoneNumber.onChange(null);
fields.phoneNumber.value === null;
```
I think this change would break that (and also empty strings)